nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+
+
+'''Method Interface'''
+
+<pre>
+function getProvider(): Promise<Provider>
+</pre>
+
+
+'''Input arguments'''
+
+None
+
+
+'''Success return value'''
+
+object - Provider information defined by the interface below:
+<pre>
+interface Provider {
+  name: string;
+  website: string;
+  version: string;
+  compatibility: string[];
+  extra: object; //this object can contain any attributes specific to the dapi provider, such as an app theme.
+}
+</pre>
+
+The compatibility field will return a list of strings that corresponds to NEPs that the dAPI conforms to. For example:
+<pre>
+compatibility: [
+    'NEP-14',
+    'NEP-23',
+    'NEP-29'
+]
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getProvider()
+.then((provider: Provider) => {
+  const {
+    name,
+    website,
+    version,
+    compatibility,
+    extra,
+  } = provider;
+
+  console.log('Provider name: ' + name);
+  console.log('Provider website: ' + website);
+  console.log('Provider dAPI version: ' + version);
+  console.log('Provider dAPI compatibility: ' + JSON.stringify(compatibility));
+  console.log('Extra provider specific atributes: ' + JSON.stringify(compatibility));
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp.');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+* The name of the provider wallet
+* The provider website
+* The provider specific version of the wallet
+* A list of NEP for which the specific provider protocol is compatible with
+
+Example
+<pre>
+{
+  name: 'Awesome Wallet',
+  website: 'https://www.awesome.com',
+  version: 'v0.0.1',
+  compatibility: [
+    'NEP-14',
+    'NEP-23',
+    'NEP-29'
+  ],
+  extra: {
+    theme: 'Dark Mode'
+  }
+}
+</pre>
+
+
+====getNetworks====
+Returns the networks the wallet provider has available to connect to, along with the default network the wallet is currently set to.
+
+
+'''Method Interface'''
+
+<pre>
+function getNetworks(): Promise<Networks>
+</pre>
+
+
+'''Input arguments'''
+
+None
+
+
+'''Success return value'''
+
+<pre>
+interface Networks {
+  networks: string[]; // Array of network names the wallet provider has available for the dapp developer to connect to.
+  defaultNetwork: string; // Network the wallet is currently set to.
+}
+</pre>
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getNetworks()
+.then(response => {
+  const {
+    networks,
+    defaultNetwork,
```

Comment:
Perhaps "default" is not a good name here... For me, default is the "standard" network,not the current one connected. If intention is return the current one,that may change as user commands, it would be better changing to "currentNetwork"

Reply:
If user cannot change this configuration, default would be a good name, but explanarions would need fixing, as default never changes, and its just standard option when user doesnt pass one.

Reply:
Can user change this value?

Reply:
Did you see my previous comment? kkkkk ahauah

Reply:
Oh! @vncoelho asked the same!! ok, not doing that.